### DISCUSSION TITLE
Signup with Oauth

### DISCUSSION THREAD

**Discussion by joshpopelka20gmail** [supabase/supabase#30755] [Questions] [answered]
I was able to implement the logic for oauth signin with google in this article: https://supabase.com/docs/guides/auth/social-login/auth-google?queryGroups=environment&environment=server

How do I implement the logic for user signup? There doesn't seem to be a default method to do this.

**Comment by GaryAustin1** [comment]
Not clear if you are asking about this... but OAuth will create the user if their email does not already exist automatically.

**Comment by joshpopelka20gmail** [comment, marked as answer]
@GaryAustin1 thanks for the reply. I did notice that. Though I was expecting a method specifically for signup. 

Now, I'm having an issue with redirects after the signup. After I figure that out, I'll close the comment.
